### ISSUE TITLE
build DM failed if GOPATH is not set

### ISSUE THREAD

**Issue by siddontang** [pingcap/dm#393]
## Question

After Go 1.2, it is not necessary to set GOPATH, but if there is not GOPATH, the DM build will fail.

Maybe we should check GOPATH is set before building, another alternative way is to only do the check in CI, or in Dev mod. 

```
make build
gofmt (simplify)
GO111MODULE=off go get golang.org/x/lint/golint
golint
bash: golint: command not found
make: *** [lint] Error 127
```

**Comment by GregoryIan** [comment]
@csuzhangxc PTAL

**Comment by csuzhangxc** [comment]
I can fix this later.

**Comment by csuzhangxc** [comment]
fixed in #446
